### PR TITLE
Wireframe

### PR DESCRIPTION
**Pull request by dlyr**
Can (should) be merged before #589.

But there is design flaws we need to discuss.
 - Where to store the wireframe meshes.
 - How to control wireframe appearance
 - Do we need better/adapted polygon offset ?

### CODE REVIEW

**Comment by dlyr** [comment]
For comparison
wireframe-old
wireframe-new

**Comment by dlyr** [comment]
The way wireframe meshes are generated and stored is not perfect.
The are stored in a map in the renderer, while it's really easy to do, it's prone to memory leaks (well there are a lot on the OpenGL side, since most of GL buffers are not deleted).
Also, the DrawPrimitive app need to be updated.

LineMesh indicies computation could also be done using topological mesh (for triangle now, for poly in a near futur), it's better in term of memory footprint (only one line per edge, guaranteed) while now we may have one line drawn multiple times.

**Comment by nmellado** [comment]
@dlyr you marked this review as ready to review, but it still have the `WIP` label. Shall we review now ?

**Comment by dlyr** [comment]
Yes, in other term, I need help and regarding previous comment.
Ready for review, not for merge ;)

**Comment by MathiasPaulin** [comment]
> The way wireframe meshes are generated and stored is not perfect.

Even if not perfect, it helps a lot and might be easily replaced later (e.g. when multiviewindex will be available). 

> The are stored in a map in the renderer, while it's really easy to do, it's prone to memory leaks (well there are a lot on the OpenGL side, since most of GL buffers are not deleted).

Even if globject lifetime management is hard to follow in the code, and after verification, all OpenGL objects constructed by globjects are automatically deleted when their owing objects got out of scope. For buffers, the documentation about the destructor clearly state that : 
```
/** Automatically deletes the associated OpenGL buffer unless the object was created with an external id.
        \see https://www.opengl.org/sdk/docs/man4/xhtml/gl::glDeleteBuffers.xml
    */
```

And no buffers in Radium is created from an already existing OpenGL buffer.

I verified using OpenGL  profiler and VAO/VBO are deleted as expected (at least on MacOs)

> Also, the DrawPrimitive app need to be updated.

Which kind of update ?

**Comment by dlyr** [comment]
Well DrawPrimitive is a bit out of sync with current RadiumApp init, or maybe I misunderstand recent changes.
Nevertheless I have pointed out an invalid read, and I got some segfault time to time...

**Comment by dlyr** [comment]
Could be done in another PR, as well as the mesh factory thing.

**Review by nmellado — APPROVED** [review]
Ready to merge if @MathiasPaulin approves.
